MapboxCoreNavigation/RouteController.swift

```diff
@@ -546,8 +567,13 @@ extension RouteController: CLLocationManagerDelegate {
         // `filteredLocations` does not contain good locations and we have found at least one good location previously.
         } else if hasFoundOneQualifiedLocation {
             if let lastLocation = locations.last, delegate?.routeController?(self, shouldDiscard: lastLocation) ?? true {
+                
                 // Allow the user puck to advance. A stationary puck is not great.
                 self.rawLocation = lastLocation
+                
+                // Check for a tunnel intersection at the current step we found the bad location update.
+                checkForTunnelIntersection(at: lastLocation, for: manager)
```

Comment:
I'd move this this out of this if statement and put it below. This function should use [`location`](https://github.com/mapbox/mapbox-navigation-ios/pull/1218/files#diff-1a2149692b3d7c58b6a0392120d6011aR585).

Reply:
@bsudekum shouldn't the simulation move along with the puck as indicated in the comment above
>  Allow the user puck to advance. A stationary puck is not great.

Reply:
@bsudekum also when you say "_I'd move this out of this if statement and put it below._"... do you mean, moving this out of the conditional statement:
` if let lastLocation = locations.last, delegate?.routeController?(self, shouldDiscard: lastLocation) ?? true` ?

Reply:
Move to at least here: https://github.com/mapbox/mapbox-navigation-ios/blob/f00ae83b54d47a7d4de7fbf3dd43ad1758a56d6b/MapboxCoreNavigation/RouteController.swift#L589

Reply:
@bsudekum It already resides here: 
https://github.com/mapbox/mapbox-navigation-ios/blob/f00ae83b54d47a7d4de7fbf3dd43ad1758a56d6b/MapboxCoreNavigation/RouteController.swift#L615

The reason the check for tunnel lives here is to ensure whenever:
> `filteredLocations` does not contain good locations and we have found at least one good location previously.

and... to follow suit as we advance the puck as indicated here:
https://github.com/mapbox/mapbox-navigation-ios/pull/1218/files#diff-1a2149692b3d7c58b6a0392120d6011aL549

... unless I'm missing something, here.